assets/js/googlesitekit/datastore/site/developer-plugin.test.js

```diff
@@ -0,0 +1,151 @@
+			it( 'requires the response param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).receiveGetDeveloperPluginState();
+				} ).toThrow( 'response is required.' );
+			} );
+
+			it( 'receives and sets developer plugin state', async () => {
+				await registry.dispatch( STORE_NAME ).receiveGetDeveloperPluginState( responseDeveloperPluginState );
+
+				const { developerPluginState } = registry.stores[ STORE_NAME ].store.getState();
+
+				expect( developerPluginState ).toMatchObject( responseDeveloperPluginState );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getDeveloperPluginState', () => {
+			it( 'uses a resolver to make a network request', async () => {
+				fetchMock.getOnce(
+					/^\/google-site-kit\/v1\/core\/site\/data\/developer-plugin/,
+					{ body: responseDeveloperPluginState, status: 200 }
+				);
+
+				const initialDeveloperPluginState = registry.select( STORE_NAME ).getDeveloperPluginState();
+				// The developer plugin state will be its initial value while the developer plugin state is fetched.
+				expect( initialDeveloperPluginState ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getDeveloperPluginState() !== undefined
+					),
+				);
+
+				const developerPluginState = registry.select( STORE_NAME ).getDeveloperPluginState();
+
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+				expect( developerPluginState ).toEqual( responseDeveloperPluginState );
+
+				const developerPluginStateSelect = registry.select( STORE_NAME ).getDeveloperPluginState();
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+
+				expect( developerPluginStateSelect ).toEqual( developerPluginState );
+			} );
+
+			it( 'does not make a network request if data is already in state', async () => {
+				registry.dispatch( STORE_NAME ).receiveGetDeveloperPluginState( responseDeveloperPluginState, {} );
+
+				const developerPluginState = registry.select( STORE_NAME ).getDeveloperPluginState();
+
+				await subscribeUntil( registry, () => registry
+					.select( STORE_NAME )
+					.hasFinishedResolution( 'getDeveloperPluginState' )
+				);
+
+				expect( fetchMock ).not.toHaveFetched();
+				expect( developerPluginState ).toEqual( responseDeveloperPluginState );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetchMock.getOnce(
+					/^\/google-site-kit\/v1\/core\/site\/data\/developer-plugin/,
+					{ body: response, status: 500 }
+				);
+
+				registry.select( STORE_NAME ).getDeveloperPluginState();
+				await subscribeUntil( registry,
+					// TODO: We may want a selector for this, but for now this is fine
+					// because it's internal-only.
+					() => registry.select( STORE_NAME ).isFetchingGetDeveloperPluginState() === false,
+				);
```

Comment:
Please use `untilResolved` instead 🙂

Reply:
@aaemnnosttv using `untilResolved` here fired `isFetchingGetDeveloperPluginState is not a function` errors.

Reply:
`untilResolved` is specific to selectors with resolvers. The above `isFetchingGetDeveloperPluginState` is a selector without a resolver which we don't need to use here. We only need to wait for `getDeveloperPluginState` to be resolved just like in the other tests.
```suggestion
				await untilResolved( registry, STORE_NAME ).getDeveloperPluginState();
```